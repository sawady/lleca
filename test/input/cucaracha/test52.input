
fun printVector(v : Vec) {
    i := 0
    putChar(91)
    while i < #v {
        putNum(v[i])
        i := i + 1
        if i < #v {
            putChar(44)
            putChar(32)
        }
    }
    putChar(93)
    putChar(10)
}

fun main() {
    v := [1, 2, 3]
    printVector(v)

    v := [10, 20, 30]
    printVector(v)

    printVector([11, 22, 33, 44, 55, 66, 77, 88, 99])
}
