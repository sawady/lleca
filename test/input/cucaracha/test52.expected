Cons(
  Function(
    printVector,
    Cons(
      Parameter(
        v,
        Vec
      ),
      Nil
    ),
    Unit,
    Cons(
      StmtAssign(
        i,
        ExprConstNum(
          0
        )
      ),
      Cons(
        StmtCall(
          putChar,
          Cons(
            ExprConstNum(
              91
            ),
            Nil
          )
        ),
        Cons(
          StmtWhile(
            ExprLt(
              ExprVar(
                i
              ),
              ExprVecLength(
                v
              )
            ),
            Cons(
              StmtCall(
                putNum,
                Cons(
                  ExprVecDeref(
                    v,
                    ExprVar(
                      i
                    )
                  ),
                  Nil
                )
              ),
              Cons(
                StmtAssign(
                  i,
                  ExprAdd(
                    ExprVar(
                      i
                    ),
                    ExprConstNum(
                      1
                    )
                  )
                ),
                Cons(
                  StmtIf(
                    ExprLt(
                      ExprVar(
                        i
                      ),
                      ExprVecLength(
                        v
                      )
                    ),
                    Cons(
                      StmtCall(
                        putChar,
                        Cons(
                          ExprConstNum(
                            44
                          ),
                          Nil
                        )
                      ),
                      Cons(
                        StmtCall(
                          putChar,
                          Cons(
                            ExprConstNum(
                              32
                            ),
                            Nil
                          )
                        ),
                        Nil
                      )
                    ),
                    Nothing
                  ),
                  Nil
                )
              )
            )
          ),
          Cons(
            StmtCall(
              putChar,
              Cons(
                ExprConstNum(
                  93
                ),
                Nil
              )
            ),
            Cons(
              StmtCall(
                putChar,
                Cons(
                  ExprConstNum(
                    10
                  ),
                  Nil
                )
              ),
              Nil
            )
          )
        )
      )
    )
  ),
  Cons(
    Function(
      main,
      Nil,
      Unit,
      Cons(
        StmtAssign(
          v,
          ExprVecMake(
            Cons(
              ExprConstNum(
                1
              ),
              Cons(
                ExprConstNum(
                  2
                ),
                Cons(
                  ExprConstNum(
                    3
                  ),
                  Nil
                )
              )
            )
          )
        ),
        Cons(
          StmtCall(
            printVector,
            Cons(
              ExprVar(
                v
              ),
              Nil
            )
          ),
          Cons(
            StmtAssign(
              v,
              ExprVecMake(
                Cons(
                  ExprConstNum(
                    10
                  ),
                  Cons(
                    ExprConstNum(
                      20
                    ),
                    Cons(
                      ExprConstNum(
                        30
                      ),
                      Nil
                    )
                  )
                )
              )
            ),
            Cons(
              StmtCall(
                printVector,
                Cons(
                  ExprVar(
                    v
                  ),
                  Nil
                )
              ),
              Cons(
                StmtCall(
                  printVector,
                  Cons(
                    ExprVecMake(
                      Cons(
                        ExprConstNum(
                          11
                        ),
                        Cons(
                          ExprConstNum(
                            22
                          ),
                          Cons(
                            ExprConstNum(
                              33
                            ),
                            Cons(
                              ExprConstNum(
                                44
                              ),
                              Cons(
                                ExprConstNum(
                                  55
                                ),
                                Cons(
                                  ExprConstNum(
                                    66
                                  ),
                                  Cons(
                                    ExprConstNum(
                                      77
                                    ),
                                    Cons(
                                      ExprConstNum(
                                        88
                                      ),
                                      Cons(
                                        ExprConstNum(
                                          99
                                        ),
                                        Nil
                                      )
                                    )
                                  )
                                )
                              )
                            )
                          )
                        )
                      )
                    ),
                    Nil
                  )
                ),
                Nil
              )
            )
          )
        )
      )
    ),
    Nil
  )
)
